scripts/firedrake-preprocess-bibtex

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+import sys
+import io
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+try:
+    from bibtexparser.bwriter import BibTexWriter
+    import bibtexparser
+except ImportError:
+    raise ImportError("Failed to import bibtexparser. Run:\n firedrake-update --documentation-dependencies")
+
+
+parser = ArgumentParser(description="""Ensure BibTeX entries for inclusion in the Firedrake website have a
+URL or DOI, and impose clean formatting.""",
+                        formatter_class=RawDescriptionHelpFormatter)
+parser.add_argument("bibtex-file", help="The BibTeX file to process")
+parser.add_argument("--validate", action='store_true',
+                    help="Instead of rewriting the bibtex file, raise an exception if anything would have changed.")
```

Comment:
It appears (check Jenkins) that option flags and positional arguments don't play well together. e.g. you can't write (with this setup) `firedrake-preprocess-bibtex --validate foo.bib`